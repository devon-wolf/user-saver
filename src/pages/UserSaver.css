.userSaver {
	height: 100%;
	width: 100%;
	display: flex;
	flex-direction: column;
	justify-content: center;
	padding: 1rem;
	background-color: lightgray;
}