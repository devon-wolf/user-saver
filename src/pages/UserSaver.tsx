import React from 'react';
import SearchForm from '../components/search/SearchForm';
import FeedbackMessage from '../components/search/FeedbackMessage';
import UserList from '../components/users/UserList';
import useUsers from '../hooks/useUsers';
import useMessage from '../hooks/useMessage';

const UserSaver = () => {
	const { users } = useUsers();
	const { message, setMessage } = useMessage();

	return (
		<div>
			<SearchForm setMessage={setMessage} />
			<FeedbackMessage message={message} />

			<UserList users={users} />
		</div>
	);
};

export default UserSaver;
