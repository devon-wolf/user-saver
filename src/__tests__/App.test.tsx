import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import UserSaver from '../pages/UserSaver';

describe('UserSaver page', () => {
  beforeEach(() => {
    render(<UserSaver />);
  });

  it('renders the user saver page', () => {
    screen.getByLabelText('search form');
    screen.getByRole('textbox');
    screen.getByRole('table');
  });

  it.skip('takes in a search term and correctly updates with results on submit', () => {
    const searchForm = screen.getByLabelText('search form');
    const searchInput = screen.getByRole('textbox');
    
    fireEvent.change(searchInput, {
      target: { value: 'devon-wolf' } 
    });

    fireEvent.submit(searchForm);

    const userTable = screen.getByRole('table');
    
  });
});



