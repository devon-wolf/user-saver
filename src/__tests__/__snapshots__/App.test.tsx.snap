// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSaver page displays a list of users and their data 1`] = `
<tr
  aria-label="user row"
>
  <td
    aria-label="username"
  >
    <a
      aria-label="link to GitHub"
      href="https://github.com/devon-wolf"
    >
      devon-wolf
    </a>
  </td>
  <td
    aria-label="name"
  >
    n/a
  </td>
  <td
    aria-label="public repos"
  >
    80
  </td>
  <td
    aria-label="public gists"
  >
    0
  </td>
  <td
    aria-label="followers"
  >
    30
  </td>
  <td
    aria-label="following"
  >
    45
  </td>
  <td
    aria-label="created at"
  >
    11/25/2017
  </td>
</tr>
`;

exports[`UserSaver page displays a list of users and their data 2`] = `
<tr
  aria-label="user row"
>
  <td
    aria-label="username"
  >
    <a
      aria-label="link to GitHub"
      href="https://github.com/Vance-M"
    >
      Vance-M
    </a>
  </td>
  <td
    aria-label="name"
  >
    n/a
  </td>
  <td
    aria-label="public repos"
  >
    55
  </td>
  <td
    aria-label="public gists"
  >
    0
  </td>
  <td
    aria-label="followers"
  >
    14
  </td>
  <td
    aria-label="following"
  >
    2
  </td>
  <td
    aria-label="created at"
  >
    11/19/2020
  </td>
</tr>
`;

exports[`UserSaver page displays a list of users and their data 3`] = `
<tr
  aria-label="user row"
>
  <td
    aria-label="username"
  >
    <a
      aria-label="link to GitHub"
      href="https://github.com/ccbass"
    >
      ccbass
    </a>
  </td>
  <td
    aria-label="name"
  >
    Colin Bassett
  </td>
  <td
    aria-label="public repos"
  >
    31
  </td>
  <td
    aria-label="public gists"
  >
    0
  </td>
  <td
    aria-label="followers"
  >
    9
  </td>
  <td
    aria-label="following"
  >
    0
  </td>
  <td
    aria-label="created at"
  >
    05/20/2020
  </td>
</tr>
`;

exports[`UserSaver page displays a list of users and their data 4`] = `
<tr
  aria-label="user row"
>
  <td
    aria-label="username"
  >
    <a
      aria-label="link to GitHub"
      href="https://github.com/phrabos"
    >
      phrabos
    </a>
  </td>
  <td
    aria-label="name"
  >
    Patrick Hrabos
  </td>
  <td
    aria-label="public repos"
  >
    92
  </td>
  <td
    aria-label="public gists"
  >
    0
  </td>
  <td
    aria-label="followers"
  >
    28
  </td>
  <td
    aria-label="following"
  >
    39
  </td>
  <td
    aria-label="created at"
  >
    10/20/2020
  </td>
</tr>
`;

exports[`UserSaver page displays a list of users and their data 5`] = `
<tr
  aria-label="user row"
>
  <td
    aria-label="username"
  >
    <a
      aria-label="link to GitHub"
      href="https://github.com/Tylerpfarris"
    >
      Tylerpfarris
    </a>
  </td>
  <td
    aria-label="name"
  >
    Tyler Farris
  </td>
  <td
    aria-label="public repos"
  >
    61
  </td>
  <td
    aria-label="public gists"
  >
    0
  </td>
  <td
    aria-label="followers"
  >
    13
  </td>
  <td
    aria-label="following"
  >
    8
  </td>
  <td
    aria-label="created at"
  >
    12/13/2020
  </td>
</tr>
`;

exports[`UserSaver page displays a list of users and their data 6`] = `
<tr
  aria-label="user row"
>
  <td
    aria-label="username"
  >
    <a
      aria-label="link to GitHub"
      href="https://github.com/dpcairns"
    >
      dpcairns
    </a>
  </td>
  <td
    aria-label="name"
  >
    Danny Cairns
  </td>
  <td
    aria-label="public repos"
  >
    401
  </td>
  <td
    aria-label="public gists"
  >
    0
  </td>
  <td
    aria-label="followers"
  >
    75
  </td>
  <td
    aria-label="following"
  >
    2
  </td>
  <td
    aria-label="created at"
  >
    12/05/2015
  </td>
</tr>
`;

exports[`UserSaver page displays a list of users and their data 7`] = `
<tr
  aria-label="user row"
>
  <td
    aria-label="username"
  >
    <a
      aria-label="link to GitHub"
      href="https://github.com/mehtaphysical"
    >
      mehtaphysical
    </a>
  </td>
  <td
    aria-label="name"
  >
    Ryan Mehta
  </td>
  <td
    aria-label="public repos"
  >
    57
  </td>
  <td
    aria-label="public gists"
  >
    12
  </td>
  <td
    aria-label="followers"
  >
    97
  </td>
  <td
    aria-label="following"
  >
    0
  </td>
  <td
    aria-label="created at"
  >
    12/09/2011
  </td>
</tr>
`;
